src/lightning_habana/pytorch/datamodule/datamodule.py

```diff
@@ -0,0 +1,210 @@
+            ):
+                print("Val Transforms supported")
+            else:
+                raise ValueError("Unsupported val transform: " + str(type(t)))
+
+        val_transforms = transform_lib.Compose(val_transforms)
+
+    if "imagenet" not in traindir.lower() and "ilsvrc2012" not in traindir.lower():
+        raise ValueError("Habana dataloader only supports Imagenet dataset")
+
+    # Data loading code
+    dataset_train = torchvision.datasets.ImageFolder(traindir, train_transforms)
+
+    dataset_val = torchvision.datasets.ImageFolder(valdir, val_transforms)
+
+    return dataset_train, dataset_val
+
+
+class HPUDataModule(pl.LightningDataModule):
+    """Datamodule helper class to load the right media pipe."""
+
+    name = "hpu-dataset"
+
+    def __init__(
+        self,
+        train_dir: str = _DATASETS_PATH,
+        val_dir: str = _DATASETS_PATH,
+        num_workers: int = 8,
+        normalize: bool = False,
+        seed: int = 42,
+        batch_size: int = 32,
+        train_transforms: Any = None,
+        val_transforms: Any = None,
+        pin_memory: bool = True,
+        shuffle: bool = False,
+        drop_last: bool = True,
+        dl_type: str = "MP",
+        distributed: bool = False,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.train_dir = train_dir
+        self.val_dir = val_dir
+        self.num_workers = num_workers
+        self.normalize = normalize
+        self.seed = seed
+        self.batch_size = batch_size
+        self.train_transform = train_transforms
+        self.val_transform = val_transforms
+        self.pin_memory = pin_memory
+        self.shuffle = shuffle
+        self.drop_last = drop_last
+        self.dl_type = dl_type
+        self.distributed = distributed
+        self.data_loader_type = torch.utils.data.DataLoader
+
+        if _HPU_AVAILABLE:
+            if lightning_habana.pytorch.datamodule.utils.is_gaudi2():
+                self.data_loader_type = MediaApiDataLoader  # type: ignore
+            else:
+                self.data_loader_type = habana_dataloader.HabanaDataLoader
+
+        if self.dl_type == "MP":
+            # users have the option to select native pytorch dataloader using "MP" type
+            self.data_loader_type = torch.utils.data.DataLoader
+
+    def setup(self, stage: Optional[str] = None):  # type: ignore[no-untyped-def]
+        """Method to sanitize the input params."""
+        if not _TORCHVISION_AVAILABLE:
+            raise ValueError("torchvision transforms not available")
+
+        if self.shuffle is True:
+            raise ValueError("HabanaDataLoader does not support shuffle=True")
+
+        if self.pin_memory is False:
+            raise ValueError("HabanaDataLoader only supports pin_memory=True")
+
+        if self.num_workers != 8:
+            raise ValueError("HabanaDataLoader only supports num_workers as 8")
```

Comment:
Doesn't this depend on the "dl_type" (see my comment above).

Reply:
its applicable only for habana data modules

Reply:
Then it should be conditioned on self.dl_type. Otherwise, if a torch DataLoader is used, the user can't change the number of workers